### ISSUE TITLE
[Improvement] Check for S3 bucket before failing

### ISSUE THREAD

**Issue by ronaldbradford** [wesql/wesql#9]
**Is your improvement request related to a problem? Please describe.**

When following your tutorial for the first time, you ask to create the S3 bucket and you do not add s3:CreateBucket as a required Action.
See: https://wesql.io/docs/tutorial/get-started

However, if you run via docker using `apecloud/wesql-server:8.0.35-0.1.0_beta1.g4a5a4eb.33` , this fails 

```
2024/11/18 14:54:01 ERROR : cluster_mgmt/cluster.json: Post request put error: AccessDenied: User: arn:aws:iam::999999999999:user/doit-wesql is not authorized to perform: s3:CreateBucket on resource: "arn:aws:s3:::doit-wesql" because no identity-based policy allows the s3:CreateBucket action
	status code: 403, request id: 3WQ6SVPW9WF5XYVP, host id: ejT6bpF7daXiiF4Yvmgri+6QnU1KqXcGF997gueEh+m0giX4/Aw8mvd7N0L7+DAGY1oB+vF15wk=
Error: push to "cluster_mgmt/cluster.json": AccessDenied: User: arn:aws:iam::9999999999:user/doit-wesql is not authorized to perform: s3:CreateBucket on resource: "arn:aws:s3:::doit-wesql" because no identity-based policy allows the s3:CreateBucket action
	status code: 403, request id: 3WQ6SVPW9WF5XYVP, host id: ejT6bpF7daXiiF4Yvmgri+6QnU1KqXcGF997gueEh+m0giX4/Aw8mvd7N0L7+DAGY1oB+vF15wk=
```

**Describe the solution you'd like**

You should check if the bucket already exists.

**Describe alternatives you've considered**

I had to delete the bucket, and add  `s3:CreateBucket` to the policy documented to get a running docker image

**Additional context**

Very cool idea, looking forward to testing it out.

**Comment by earayu** [comment]
Hi, @ronaldbradford 
Thank you for bringing this to our attention. You are right, `apecloud/wesql-server:8.0.35-0.1.0_beta1.g4a5a4eb.33` requires the `s3:CreateBucket` permission.

We will soon release a minor version update that removes the dependency on the `s3:CreateBucket` permission.

In the meantime, you can get a free S3 Bucket for testing purposes at https://wesql.io/bucket.

Thanks again for your feedback!

**Comment by demian0110** [comment]
This issue has been resolved in apecloud/wesql-server:8.0.35-0.1.0_beta3.38. Please re-verify.

**Comment by ronaldbradford** [comment]
I can confirm that having existing bucket works with apecloud/wesql-server:8.0.35-0.1.0_beta3.38
